src/marqo/index.py

```diff
@@ -85,10 +93,21 @@ def create(config: Config, index_name: str,
         Returns:
             Response body, containing information about index creation result
         """
+        def cloud_wait_for_index_ready():
```

Comment:
Let's not use nested functions where possible. Perhaps we create file `cloud_helpers.py` for these types of functions. Then we can also create unit tests for these functions